inc/cookie/namespace.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace HM\Delegated_Auth\Cookie;
+
+use WP_Error;
+use function HM\Delegated_Auth\synchronize_user_for_token;
+
+function is_enabled() : bool {
+	return defined( 'HM_DELEGATED_AUTH_CLIENT_ID' );
+}
+/**
+ * Login page footer for the login link.
+ */
+function on_login_form() : void {
+	?>
+	<p><a href="<?php echo esc_url( get_authorize_url() ); ?>">Login with Delegated Auth</a></p>
+	<?php
+}
+
+function on_load() {
+	if ( wp_parse_url( $_SERVER['REQUEST_URI'], PHP_URL_PATH ) !== '/hm-delegated-auth-callback' ) {
+		return;
+	}
+
+	$code = sanitize_text_field( $_GET['code'] );
+	$auth = on_auth_callback( $code );
+	if ( is_wp_error( $auth ) ) {
+		wp_die( $auth->get_error_message() );
+	}
+}
+
+function on_auth_callback( string $code ) {
+	$args = [
+		'client_id'    => HM_DELEGATED_AUTH_CLIENT_ID,
+		'redirect_uri' => home_url( '/hm-delegated-auth-callback' ),
+		'grant_type'   => 'authorization_code',
+		'code'         => $code,
+	];
+	$response = wp_remote_post( HM_DELEGATED_AUTH_REST_BASE . 'oauth2/access_token', [
+		'body' => $args,
+	] );
+	if ( is_wp_error( $response ) ) {
+		return $response;
+	}
+
+	$body = json_decode( wp_remote_retrieve_body( $response ), true );
+
+	if ( wp_remote_retrieve_response_code( $response ) !== 200 ) {
+		return new WP_Error( 'invalid-access-token', sprintf( 'Invalid access token, recieved %s (%s)', $body['message'], $body['code'] ) );
```

Comment:
Misspelled received. Should also check `json_last_error()` against `JSON_ERROR_NONE`